src/anyon/anyon.jl

```diff
@@ -374,31 +374,179 @@ end
 
 const GeometricConnectivity = Union{LineConnectivity,LatticeConnectivity}
 
-function is_native_instruction(gate::Gate, connectivity::GeometricConnectivity)::Bool
-    if any(x -> x in connectivity.excluded_positions, get_connected_qubits(gate))
+"""
+    is_native_instruction(
+        gate::Union{Gate},
+        connectivity::Union{LineConnectivity,LatticeConnectivity},
+        native_gates::Vector{DataType} = set_of_native_gates,
+    )::Bool
+
+Returns `true` if the `gate` is a native instruction for the `connectivity` and the list of
+possible `native_gates`. The native gates for the Anyon QPUs are used by default.
+
+A native instruction is defined as an instruction that is in `native_gates` and that
+satisifies the `connectivity`. It does not check to determine if the `gate` is placed at
+the `excluded_positions` or `excluded_connections` of the `connectivity`.
+
+# Example
+
+```jldoctest  
+julia> connectivity = LineConnectivity(3)
+LineConnectivity{3}
+1──2──3
+
+
+julia> is_native_instruction(control_z(1, 2), connectivity)
+true
+
+julia> is_native_instruction(control_z(1, 3), connectivity)
+false
+
+julia> is_native_instruction(toffoli(1, 2, 3), connectivity)
+false
+
+julia> is_native_instruction(toffoli(1, 2, 3), connectivity, [Snowflurry.Toffoli])
+true
+
+```
+"""
+function is_native_instruction(
+    gate::Gate,
+    connectivity::GeometricConnectivity,
```

Comment:
let's also implement `is_native_instruction` for `AllToAllConnectivity` (all gates are native)